docker/build-tool.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# Ensure that commands are run from this directory.
+cd "$(dirname "${0}")"
+
+HELP_TEXT="$(cat <<EOF
+
+ Builds a tool docker image.
+   -p | --push : (default: only build) if set, pushes the image after build.
+           If unset, only perform a build.
+   -i | --image : the tool image name.
+   -r | --repository : the docker repository to push to.  Can be a dockerhub or GCR repo.
+   -t | --tag : the docker tag used for the images.
+   -b | --branch : (default: none) the git branch
+   -h | --help: print help text.
+
+ Examples:
+   Builds and pushes the Jupyter docker image
+    ./docker/build-tool.sh --push -i jupyter -r us.gcr.io/broad-dsp-gcr-public -t 12345 -b my-branch
+   Builds the RStudio docker image tagged with the git branch name
+    ./docker/build-tool.sh -i rstudio -r us.gcr.io/broad-dsp-gcr-public -t my-tag
+\t
+EOF
+)"
+
+# Enable strict evaluation semantics.
+set -e
+
+# Set default variables used while parsing command line options.
+PRINT_HELP=false
+PUSH=false
+IMAGE=""
+REPOSITORY=""
+DOCKER_TAG=""
+BRANCH=""
+
+if [ -z "$1" ]; then
+    echo "No argument supplied!"
+    echo "run '${0} -h' to see available arguments."
+    exit 1
+fi
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -p | --push)
+            echo "will perform a PUSH"
+            PUSH=true
+            ;;
+        -i | --image)
+            shift
+            IMAGE=$1
+            ;;
+        -r | --repository)
+            shift
+            REPOSITORY=$1
+            ;;
+        -t | --tag)
+            shift
+            DOCKER_TAG=$1
+            ;;
+        -b | --branch)
+            shift
+            BRANCH=$1
+            ;;
+        -h | --help)
+            PRINT_HELP=true
+            ;;
+        *)
+            echo "Unrecognized argument '${1}'."
+            echo "run '${0} -h' to see available arguments."
+            if grep -Fq "=" <<< "${1}"; then
+                echo "note: separate args from flags with a space, not '='."
+            fi
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ -z "$IMAGE" ]; then
+    echo "image is required"
+    PRINT_HELP=true
+fi
+
+if [ -z "$REPOSITORY" ]; then
+    echo "repository is required"
+    PRINT_HELP=true
+fi
+
+if [ -z "$REPOSITORY" ]; then
+    echo "tag is required"
+    PRINT_HELP=true
+fi
+
+# Print help after all flags are parsed successfully
+if $PRINT_HELP; then
+  echo -e "${HELP_TEXT}"
+  exit 0
+fi
+
+GIT_BRANCH="${BRANCH:-$(git rev-parse --abbrev-ref HEAD)}"
+
+# for backwards compatibility; jupyter images are named 'leonardo-notebooks'
```

Comment:
is there a plan to drop this legacy name at some point?

Reply:
I included this because `leonardo-notebooks` is specified in firecloud-develop `leonardo.conf` and I wanted to avoid a coupled leonardo and firecloud-develop merge. 

Hm, a way to drop this legacy name would be to:
1. temporary push the image to BOTH `leonardo-notebooks` and `leonardo-jupyter` in the Leo build
2. update firecloud-develop to reference `leonardo-jupyter`
3. stop pushing to `leonardo-notebooks`

I can look into making that change.

Reply:
I did this, testing now